umh-core/cmd/main.go

```diff
@@ -141,33 +191,31 @@ func SystemSnapshotLogger(ctx context.Context, controlLoop *control.ControlLoop,
 	}
 }
 
-func enableBackendConnection(config *config.FullConfig, state *fsm.SystemSnapshot, communicationState *communication_state.CommunicationState, systemMu *sync.Mutex) {
+func enableBackendConnection(configData *config.FullConfig, systemSnapshot *fsm.SystemSnapshot, communicationState *communication_state.CommunicationState, systemMu *sync.Mutex) {
 	logger := logger.For("enableBackendConnection")
 	if logger == nil {
 		logger = zap.NewNop().Sugar()
 	}
 
-	logger.Info("Enabling backend connection")
-	// directly log the config to console, not to the logger
-	if config == nil {
-		logger.Warn("Config is nil, cannot enable backend connection")
+	// Check if the backend communication should be enabled (APIURL and AuthToken must be set)
+	if configData.Agent.CommunicatorConfig.APIURL == "" || configData.Agent.CommunicatorConfig.AuthToken == "" {
+		logger.Info("Backend communication disabled - missing APIURL or AuthToken")
```

Comment:
"error: Potential nil panic detected. Observed nil flow from source to dereference point: "
--> need to chec kfirst if configData is not nil, then if agent is not nil, etc. etc.